src/components/calculator.js

```diff
@@ -1,36 +1,63 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 import './calculator.css';
 
-function Calculator() {
-  return (
-    <div className="container">
-      <div className="calc-grid">
-        <div className="output">
-          <div className="current-operand">0</div>
-        </div>
-        <button type="button">AC</button>
-        <button type="button">+/-</button>
-        <button type="button">%</button>
-        <button type="button" className="orange">÷</button>
-        <button type="button">7</button>
-        <button type="button">8</button>
-        <button type="button">9</button>
-        <button type="button" className="orange">*</button>
-        <button type="button">4</button>
-        <button type="button">5</button>
-        <button type="button">6</button>
-        <button type="button" className="orange">-</button>
-        <button type="button">1</button>
-        <button type="button">2</button>
-        <button type="button">3</button>
-        <button type="button" className="orange">+</button>
-        <button type="button" className="span-two">0</button>
-        <button type="button">.</button>
-        <button type="button" className="orange">=</button>
-      </div>
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: '0',
+      next: null,
+      operation: null,
+    };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
 
-    </div>
-  );
+ clickHandler = (e) => {
+   const data = calculate(this.state, e.target.innerText);
+   this.setState(data);
+ };
+
+ render() {
+   const { total, operation, next } = this.state;
+   return (
+     <div className="container">
+       <div className="calc-grid">
+         <div className="output">
+           {`${total || ''} ${operation || ''} ${next || ''}`}
+           {/* <div className="current-operand">0</div> */}
+         </div>
+         <div
+           className="input"
+           role="button"
+           tabIndex={0}
+           onClick={this.clickHandler}
+           onKeyDown={this.clickHandler}
+         >
+           <button type="button">AC</button>
+           <button type="button">+/-</button>
+           <button type="button">%</button>
+           <button type="button" className="orange">÷</button>
+           <button type="button">7</button>
+           <button type="button">8</button>
+           <button type="button">9</button>
+           <button type="button" className="orange">*</button>
```

Comment:
- Kindly note that according to [this condition](https://github.com/myelin0/Math-macicians/pull/3/files#diff-73a54f78cf3c7060c7a99324d028518d0e55814b5e4a5bafd97bae966a1e6591R12-R14) in the `operate.js` file, the operation symbol should be `x` and not `*`.
I was not able to multiply two numbers because of an `Unknown Operation *` error.

I recorded a GIF 👇 to further show you what the error looks like.

![GIF](https://m.imgur.com/XzBomMi.gif)

Kindly fix this. 😉

Reply:
Thanks for your tip!:)